lib/yfom/value/osm_geocode.php

```diff
@@ -322,4 +337,75 @@ protected function validateNoDb(string $field_name, ?int $value, bool $return, r
         }
         return false;
     }
+
+    /**
+     * Validator für die Feld-Konfiguration
+     * 
+     * Überprüft, ob die Angaben zur Kartenformatierung (height bzw. format) korrekt sind
+     * - nur eines der beiden Felder darf gefüllt sein
+     * - die Höhe (height) muss eine Integer-Zahl sein, ggf. mit einer Massangabe
+     * - die Mssangabe darf sein px, em, rm oder vh. px ist default
+     * - wenn kein Feld gefüllt ist, wird 400px als Höhe eingetragen (default)
+     * - Die Eingabe darf mit Leerzeichen sein, die hier entfernt werden.
+     *  
+     * @param string[] $field_name
+     * @param array<string, string> $value
+     * @param array<rex_yform_value_abstract> $fields
+     */
+    protected function validateLayout(array $field_name, array $value, bool $return, rex_yform_validate_customfunction $self, array $fields): bool
+    {
+        /**
+         * Ist überhaupt in beiden Feldern etwas angegeben? Wenn nein: Höhe auf 400px setzen
+         */
+        $value = array_map(trim(...),$value);
+        $value = array_filter($value,strlen(...));
+        if( 0 === count($value)) {
+            $fields['height']->setValue('400px');
+            return false;
+        }
+
+        /**
+         * Beide Werte sind angegeben -> Meckern
+         */
+        if( 2 === count($value)) {
+            $self->setElement(
+                'message',
+                'Bitte nur die «Map-H&ouml;he»  oder die «CSS-Klasse» angeben!',
+            );
+            return true;
+        }
+
+        /**
+         * Die CSS-Klasse als einziges Feld wird nicht weiter analysiert
+         * Wert getrimmed setzen
+         */
+        if( 'format' === key($value)) {
+            $fields['format']->setValue($value['format']);
+            return false;
+        }
+
+        /**
+         * Die Höhe muss ein reiner Integer-Wert sein oder ein Int-Wert mit Masseinheit.
+         * px, em, rem, vh
+         */
+        $ok = preg_match('@^(?<height>[1-9]\d*)\s*(?<unit>px|em|rem|vh)?$@',$value['height'],$match);
+        if( 0 === $ok) {
+            $self->setElement(
+                'message',
+                sprintf(
+                    'Ungültige Eingabe «%s»; bitte nur eine Integer-Zahl sowie eine der zulässigen Masseinheiten eingeben',
```

Comment:
Hier bereits in lang-Datei auslagern?

Reply:
Ich sach mal: nö. Ich wüde einen separaten PR nur dafür machen, korrespondierend zu Issue #37